src/webots/engine/WbSimulationCluster.cpp

```diff
@@ -638,20 +646,31 @@ void WbSimulationCluster::odeNearCallback(void *data, dGeomID o1, dGeomID o2) {
     }
   }
 
-  dContact contact[MAX_CONTACTS];
-  int n = dCollide(o1, o2, MAX_CONTACTS, &contact[0].geom, sizeof(dContact));
+  const int maxContactJoints = MAX_CONTACT_JOINTS;
+  thread_local std::vector<dContact> contactVector;
```

Comment:
Similarly, we tend to use the Qt interface to std::vector, e.g., QVector elsewhere in Webots.
Do you have any reason for not doing the same here?